### ISSUE TITLE
Deprecation of `computed` throws warnings, will break stuff in Ember 3.5.0 

### ISSUE THREAD

**Issue by psteininger** [simplabs/ember-validated-form-buffer#126]
I am seeing deprecations regarding the use of `computed` in `property.js`. https://www.emberjs.com/deprecations/v3.x#toc_ember-meta-descriptor-on-object

I am not entirely sure how to address this.

**Comment by psteininger** [comment]
it's these lines, that are causing the warning:
https://github.com/simplabs/ember-validated-form-buffer/blob/master/addon/property.js#L20-L24

**Comment by marcoow** [comment]
Hm, I'm a bit surprised by this tbh - there should be no difference really whether you do

```js
prop: computed(…)
```

or

```js
function macro() {
  return computed(…);
}

{
  prop: macro()
}
```

I wonder whether this is a false positive of the deprecation.

**Comment by psteininger** [comment]
@marcoow I got rid of the warning by changing how the property is set up:
instead of:
```js
init() {
    this._super(...arguments);
    this.changeset = formBufferProperty('model', get(this, 'modelValidations'));
}
```
use this:
```js
 init() {
    this._super(...arguments);
    defineProperty(this, 'changeset', formBufferProperty('model', get(this, 'modelValidations')));
  }
```
So, I would just  update the documentation, and it's all set.

**Comment by marcoow** [comment]
Ah, I see - yes, this is what the deprecation is about:

```js
this.changeset = formBufferProperty(…
```

The suggested way of defining properties like that is this though:

```js
Ember.Object.extend({
  changeset: formBufferProperty(…
})
```

**Comment by psteininger** [comment]
@marcoow the suggested way works most of the time, except when using in a mixin. for some reason, I would get errors about missing property 'model' if I made it an outright property of the mixin. that's why it needed to be in `init()`. By updating the docs, I meant, including this edge case, for when the declaration needs to happen at "runtime".
